### PR TITLE
Limit the inline nlsolve size

### PR DESCRIPTION
**Pull request by YingboMa**
This should alleviate much of ODAEProblem compile time blow up.

### CODE REVIEW

**Comment by YingboMa** [comment]
Oh wait. There are some bugs with this.

File: src/structural_transformation/codegen.jl [hunk 1]
```diff
@@ -184,41 +188,54 @@ function gen_nlsolve(eqs, vars, u0map::AbstractDict; checkbounds=true)
...
 
+    s = structure(sys)
+    @unpack fullvars, partitions = s
+
+    states = map(i->s.fullvars[i], diffvars_range(s))
+    mass_matrix_diag = ones(length(states))
+    torn_expr = []
+    defs = defaults(sys)
+
+    needs_exdending = false
```
Comment (ChrisRackauckas) [inline src/structural_transformation/codegen.jl:215]: extending? haha

Comment (YingboMa) [reply src/structural_transformation/codegen.jl:215]: Oof, that was a mistake.

File: src/structural_transformation/codegen.jl [hunk 2]
```diff
@@ -184,41 +188,54 @@ function gen_nlsolve(eqs, vars, u0map::AbstractDict; checkbounds=true)
...
+            append!(torn_expr, gen_nlsolve(torn_eqs, torn_vars, defs, checkbounds=checkbounds))
+        else
+            needs_extending = true
+            append!(rhss, map(x->x.rhs, torn_eqs))
+            append!(states, torn_vars)
+            append!(mass_matrix_diag, zeros(length(torn_eqs)))
+        end
+    end
+
+    mass_matrix = needs_exdending ? Diagonal(mass_matrix_diag) : I
```
Comment (bolognam) [inline src/structural_transformation/codegen.jl:230]: Oof here too `needs_extending`

Comment (YingboMa) [reply src/structural_transformation/codegen.jl:230]: Oof

**Pull request merged**